deblur/workflow.py

```diff
@@ -382,40 +347,133 @@ def write_biom_table(table, biom_fp):
     biom_fp: string
         filepath to output BIOM table
     """
+    logger = logging.getLogger(__name__)
+    logger.debug('write_biom_table to file %s' % biom_fp)
     with biom_open(biom_fp, 'w') as f:
         if HAVE_H5PY:
+            logger.debug('saving with h5py')
             table.to_hdf5(h5grp=f, generated_by="deblur")
+            logger.debug('wrote to hdf5 file %s' % biom_fp)
         else:
+            logger.debug('no h5py. saving to json')
             table.to_json(direct_io=f, generated_by="deblur")
+            logger.debug('wrote to json file %s' % biom_fp)
+
+
+def get_files_for_table(input_dir,
+                        file_end='.fasta.trim.derep.no_artifacts.msa.deblur.no_chimeras'):
+    """Get a list of files to add to the output table
+
+    Parameters:
+    -----------
+    input_dir : string
+        name of the directory containing the deblurred fasta files
+    file_end : string
+        the ending of all the fasta files to be added to the table
+        (default '.fasta.trim.derep.no_artifacts.msa.deblur.no_chimeras')
+
+    Returns
+    -------
+    names : list of tuples of (string,string)
+        list of tuples of:
+            name of fasta files to be added to the biom table
+            sampleid (file names without the file_end and path)
+    """
+    logger = logging.getLogger(__name__)
+    logger.debug('get_files_for_table input dir %s, '
+                 'file-ending %s' % (input_dir, file_end))
 
+    names = []
+    # scan the input directory entries
+    for cfile in listdir(input_dir):
```

Comment:
so this does have a practical limit of 30k samples due to stress on the filesystem... but that can be revisited in a later PR. note, metadata checks (like `isfile` and `stat`) will get appreciably expensive and stressful to the filesystem under a large number of samples, but again, worth deferring to later